src/Neo/SmartContract/Manifest/ContractEventDescriptor.cs

```diff
@@ -77,5 +77,12 @@ public virtual JObject ToJson()
             json["parameters"] = new JArray(Parameters.Select(u => u.ToJson()).ToArray());
             return json;
         }
+
+        public bool Equals(ContractEventDescriptor other)
```

Comment:
You have to override `Equals(object obj)` and `GetHashCode` for this to work correctly.